### DISCUSSION TITLE
issue: The  value from Controller's render  is undefined when  use multi-select in useFieldArray

### DISCUSSION THREAD

**Discussion by wyyufeng** [react-hook-form/react-hook-form#11152] [Q&A]
### Version Number

7.47.0

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-usefieldarray-forked-532st9?file=/src/index.js

### Steps to reproduce

1. Click "append" to append a form item
2. Click  'setValue' to set the value
3. the value form  the Controller's render is undefined 

### Expected behaviour

the value is an array 

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Moshyfawn** [comment]
What's your use-case for overriding the `append`ed data with a manual `setValue`? The `append`ed data [should not be partial](https://react-hook-form.com/docs/usefieldarray#:~:text=Important%3A%20append%20data%20is%20required%20and%20not%20partial.)

```diff
- append({ selection: [] })
+ append({ businessTypes: ["IMP", "SAL"], name: "Jack", age: 123, selection: [] })
```

**Comment by wyyufeng** [comment]
Thank you for your reply！Let me explain my use case:
I try to  implement an editable table with `useFieldArray`. and the steps is:
1. click the 'append' button to add a blank row to the table,(call `append`)
2. then use the firset cell with a Select  to search some id
3. then get  the rest data by id  from server ,and call `setValue(`data.0`,data)` to fill the rest cell
